### DISCUSSION TITLE
Not getting GPS Signal Lock

### DISCUSSION THREAD

**Discussion by KJ5CMF** [kg4vdk/arcos-linux-modules#47] [General Discussion]
Has anyone had issues getting a GPS lock? I have yet to get one in Arc so far. I have a cheap U-block 7 (I believe) USB GPS receiver. I thought at first maybe it was due to being inside, but I tested today outside with no success. To get a lock in Arc what I have had to do is boot up in Windows, wait for a GPS lock, then reboot into Arc without touching the GPS, and then Arc will recognize it. Has anyone else run into this? I will say I have not let it sit for more than 5min or so, but windows is getting a lock in about a min or less.

**Comment by kg4vdk** [comment]
Some clarifying questions...

- Is this a newly purchased GPS device, or one you've had and been using recently?
- Can you provide the output of the terminal command `lsusb` (specifically, the line referring to your GPS)?
- When you say, "boot up in Windows...reboot into Arc...and then Arc will recognize it", does that mean prior to that arcOS reported "No GPS Device" in the Station Info section in the lower right of the desktop even with the GPS attached?
- Once recognized in arcOS, is the GPS seeing satellites? While waiting for a fix in arcOS, open a terminal window, and run the `cgps` command. Are satellites populating the right-hand pane of the window? 
- What software are you using in Windows for the GPS?

**Reply by KJ5CMF** [reply]
-     Is this a newly purchased GPS device, or one you've had and been using recently? - I have had this unit for several months, but just now noticing this in ArcOS
-     Can you provide the output of the terminal command lsusb (specifically, the line referring to your GPS)? - Screenshot provided

-     When you say, "boot up in Windows...reboot into Arc...and then Arc will recognize it", does that mean prior to that arcOS reported "No GPS Device" in the Station Info section in the lower right of the desktop even with the GPS attached? - I should have worded that better. It says waiting for lock. Instead of saying recognize I should have said that it gets a GPS lock in Arcos. I have provided a screenshot as well.

-     Once recognized in arcOS, is the GPS seeing satellites? While waiting for a fix in arcOS, open a terminal window, and run the cgps command. Are satellites populating the right-hand pane of the window? - Screenshot provided

-     What software are you using in Windows for the GPS? - Not using any particular software in this instance. I just boot up into windows, wait for the GPS dongle to go from red to red/blinking green, indicating a lock, then I reboot into ArcOS, and after setting up my station, seem to have a GPS lock. The windows software that I typically use though with a GPS is WinTAK and YAAC.
![Screenshot from 2024-12-12 19-44-37](https://github.com/user-attachments/assets/9facf7e5-f503-40cc-94c3-c82f78493387)
![Screenshot from 2024-12-12 19-45-23](https://github.com/user-attachments/assets/7e9d660e-f963-4ae9-90b7-4be9af787719)
![Screenshot from 2024-12-12 19-45-56](https://github.com/user-attachments/assets/cbda8211-915a-405c-bd7c-a8d55c8ad2b4)

**Reply by kg4vdk** [reply]
Thanks for the information! I have a GPS module that reports with the same hardware information (ID 1546:01a7 U-Blox AG [u-blox 7]). I can at least confirm that it works as expected in my testing.

Take "as expected" with a grain of salt, though. I just grabbed mine (one that hasn't seen the sky while plugged in for a while), and booted up with it plugged in on the hood of the truck outside.

The symptoms you describe sound very similar to my current test of a "cold start". We are using "pure" GPS, as opposed to A[ssisted]-GPS. A-GPS, as used in phones, can cheat by downloading the GPS almanac data over a network connection. Unfortunately, we have to wait until the almanac is downloaded by the GPS module directly from the satellites. On a unit which has been powered off for some time, moved significantly since its last fix, or otherwise has a stale almanac, getting this data can take a long time. The time to get the almanac data, and then a fix, can vary based on how obstructed the sky view is, as well as which satellites in the constellation are visible at that time.

Some good information here:
https://en.wikipedia.org/wiki/Time_to_first_fix
https://gpsd.gitlab.io/gpsd/faq.html#startup

My test just wrapped up, and it took about 20 minutes to finally get a location fix. I've found that the ublox-7 modules are a bit on the "deaf" side, so unobstructed sky is paramount. After my test, I powered off the laptop, unplugged the GPS, and waited about 5 minutes. Re-running the test, I had a location fix in a little under 2 minutes (as reported in `cgps`).

Hope this helps!
Mike

**Reply by KJ5CMF** [reply]
Mike, got it, thank you for the info! I'll have to do some more testing.

Taylor

**Reply by kg4vdk** [reply]
In the truck I use a GlobalStar BU-353-S4, and find that it usually has a location fix before I even get to the Station Setup window. A bit pricier (~$45USD), but the performance difference is undeniable. There is an updated version of it now, the BU-353-N5, but I have not personally tested it. 

Another user, Chuck, opened a [discussion](https://github.com/kg4vdk/arcos-linux-modules/discussions/16) regarding the updated version, and mentioned an issue with the baud rate. He got it working, however.

**Reply by KJ5CMF** [reply]
I just tested with another laptop and another USB stick (same model and brand as the first) and it worked! I just plugged in the original one I was having an issue with and if also has a lock now too. I am in my truck in a very open area. I took a chance and ordered a BU-353-N5 last night to try, should get it today. Hoping that works, and if not I'll do some digging on my end too. Thanks again for the info
![Screenshot from 2024-12-13 15-53-40](https://github.com/user-attachments/assets/9aa17319-3c79-4dcd-a694-179a8efba278)
![Screenshot from 2024-12-13 15-54-44](https://github.com/user-attachments/assets/08bb0741-6ec2-4858-a73a-52bd4bd58b77)

**Reply by kg4vdk** [reply]
Glad to hear it! And, I look forward to the test results with the BU-353-N5!
